packages/js/src/core/fetch-token.test.ts

```diff
@@ -1,19 +1,54 @@
+import { fetchTokenByAuthorizationCode } from '.';
 import { fetchTokenByRefreshToken, RefreshTokenTokenResponse } from './fetch-token';
 
+describe('fetch access token by providing authorization code', () => {
+  test('should return access token', async () => {
+    const mockedOidcResponse = {
+      access_token: 'access_token_value',
+      refresh_token: 'refresh_token_value',
+      id_token: 'id_token_value',
+      scope: 'read register',
+      expires_in: 3600,
+    };
+    const expectedTokenResponse: RefreshTokenTokenResponse = {
+      accessToken: 'access_token_value',
+      refreshToken: 'refresh_token_value',
+      idToken: 'id_token_value',
+      scope: 'read register',
+      expiresIn: 3600,
+    };
+
+    const fetchFunction = jest.fn().mockResolvedValue(mockedOidcResponse);
+
+    const tokenResponse = await fetchTokenByAuthorizationCode(
+      {
+        clientId: 'client_id_value',
+        tokenEndpoint: 'https://logto.dev/oidc/token',
+        redirectUri: 'https://localhost:3000/callback',
+        codeVerifier: 'code_verifier_value',
+        code: 'code_value',
+        resource: 'resource_value',
+      },
+      fetchFunction
+    );
+    expect(tokenResponse).toEqual(expectedTokenResponse);
+  });
+});
+
 describe('fetch access token by providing valid refresh token', () => {
-  test('get token response', async () => {
+  test('should return access token', async () => {
```

Comment:
should `return token` (more than access token).
Also, line 38.

<img width="727" alt="image" src="https://user-images.githubusercontent.com/10594507/152931587-c6af4452-8bc7-4e84-8c86-a6cd66ed602c.png">

Reply:
improved